### DISCUSSION TITLE
Export tabular output from eda to csv

### DISCUSSION THREAD

**Discussion by sgapple** [autogluon/autogluon#3149] [Q&A] [answered]
Is there any way to export the output from eda to csv? example: the covariate_shift_detection  and feature importance tabular output. I was able to do it with TabularPredictor, however, can't find any documentation for eda module to export the tabular to csv. Kind advise.

**Comment by gradientsky** [comment, marked as answer]
Some of the data is stored in state. I recently added more documentation in [dev version of the docs](https://auto.gluon.ai/dev/tutorials/eda/components/autogluon.eda.shift.html#module-autogluon.eda.analysis.shift) - see `State attributes` section.
```python
state = auto.covariate_shift_detection(
    train_data=df_train, 
    test_data=df_test,
    label=label,
    return_state=True,  # return state
)

# Get feature importance data
covariate_shift_feature_importance = state.xshift_results.feature_importance
```

**Comment by sgapple** [comment]
Thank you. the EDA has being very useful for quick data analysis.
